addons/base_import/models/base_import.py

```diff
@@ -637,6 +644,28 @@ def _parse_import_data_recursive(self, model, prefix, data, import_fields, optio
                 # We should be able to manage both case
                 index = import_fields.index(name)
                 self._parse_float_from_data(data, index, name, options)
+            elif field['type'] == 'binary' and field.get('attachment') and ('image' in name or 'icon' in name or 'logo' in name) and name in import_fields:
+                if not self.env.user._can_import_remote_urls():
+                    raise AccessError(_("Only administrators can import images via URL"))
+
+                index = import_fields.index(name)
+                for num, line in enumerate(data):
+                    if re.match(url_regex, line[index]):
+                        try:
+                            with requests.Session() as session:
+                                response = session.get(line[index], stream=True, timeout=request_timeout)
+                                chunk_size = 0
+                                content = bytearray()
+                                response.raise_for_status()
+                                for chunk in response.iter_content(DEFAULT_IMAGE_CHUNK_SIZE):
+                                    chunk_size += len(chunk)
+                                    if chunk_size > maxsize:
+                                        raise ValueError(_("File size exceeds configured maximum (%s bytes)") % maxsize)
+                                    content += chunk
```

Comment:
Why not first concatenate the new chunk to the buffer, then check if its length is higher than expected? That would avoid the need for a separate counter.